modules/tensor_mechanics/src/materials/ComputeIsotropicLinearElasticPFFractureStress.C

```diff
@@ -119,26 +114,75 @@ ComputeIsotropicLinearElasticPFFractureStress::computeQpStress()
     _hist[_qp] = _hist_old[_qp];
 
   // Damage associated with positive component of stress
-  _stress[_qp] =
-      stress0pos * cfactor * ((1.0 - c) * (1.0 - c) * (1 - _kdamage) + _kdamage) - stress0neg;
+  _stress[_qp] = stress0pos * ((1.0 - c) * (1.0 - c) * (1 - _kdamage) + _kdamage) - stress0neg;
 
   // Elastic free energy density
-  _F[_qp] = _hist[_qp] * cfactor * ((1.0 - c) * (1.0 - c) * (1 - _kdamage) + _kdamage) - G0_neg +
+  _F[_qp] = _hist_old[_qp] * ((1.0 - c) * (1.0 - c) * (1 - _kdamage) + _kdamage) - G0_neg +
```

Comment:
I'm not sure we want to hard code this to always use the old version of the history variable. Shuaifang has seen some results where it gives strain results using the old version of the history variable. Maybe make it an option of the user to choose to use old or current.

Reply:
In my opinion, I think we should always use the old value and not the current one since the history variable is defined as max[0-t_n]. If we update this variable during the current iteration, it should not be called 'history' variable anymore. Actually most of paper use staggered approach and they automatically use the old value for this.

The use of the current value also makes convergence bad. One more thing, if we want to use ACKernel, the derivatives of this term respect to displacement cannot be automatically added to off-diagonal Jacobian.